In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
!pip install torchviz

  Preparing metadata (setup.py) ... done
  Created wheel for torchviz: filename=torchviz-0.0.2-py3-none-any.whl size=4131 sha256=8b76e3f10acf35a57a55189aeff584f8c7d86df045cff22a9afac19164c836a8
  Stored in directory: /root/.cache/pip/wheels/4c/97/88/a02973217949e0db0c9f4346d154085f4725f99c4f15a87094
Successfully built torchviz


In [3]:
%cd /content/drive/MyDrive/DL_Stat/NFG-VAE

/content/drive/MyDrive/DL_Stat/NFG-VAE


In [4]:
import torch
import torch.nn as nn
import torch.nn.functional as F
import torch.optim as optim
from torchviz import make_dot
import numpy as np
import matplotlib.pyplot as plt
import os
import argparse

from model import *

# Node 개수 : 10

In [ ]:
# SEED = 1

## IAF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=100, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299: 100% 30000/30000 [3:09:34<00:00,  3.13it/s, nll_train=0.9308, kl_train=0.0504, ELBO_loss=0.9812, shd_trian=23.0000, h_A=0.0006]Optimization Finished!
results/independence/normal/1127_1223_IAF_node10_seed1"""
## HF
"nan"
## DAGGNN
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=100, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299:  27% 8100/30000 [42:01<2:17:19,  2.66it/s, nll_train=0.0001, kl_train=0.0003, ELBO_loss=0.0003, shd_trian=4.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1127_1629_DAGGNN_node10_seed1"""

# SEED = 2

## IAF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=2, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299:  38% 5700/15000 [44:17<1:03:12,  2.45it/s, nll_train=0.0000, kl_train=0.0001, ELBO_loss=0.0001, shd_trian=9.1200, h_A=0.0000]Optimization Finished!
results/independence/normal/1128_0736_IAF_node10_seed2"""
## HF
"nan"
## DAGGNN
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=2, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299:  50% 7500/15000 [36:38<31:58,  3.91it/s, nll_train=0.0000, kl_train=0.0001, ELBO_loss=0.0001, shd_trian=1.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1128_1106_DAGGNN_node10_seed2"""

# SEED = 3

## IAF
"nan"
## HF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=3, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [1:53:36<00:00,  2.55it/s, nll_train=0.6193, kl_train=0.0547, ELBO_loss=0.6740, shd_trian=27.0000, h_A=0.0017]Optimization Finished!
results/independence/normal/1128_1152_HF_node10_seed3"""
## DAGGNN
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=3, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299:  56% 8400/15000 [43:02<33:17,  3.30it/s, nll_train=0.0001, kl_train=0.0002, ELBO_loss=0.0003, shd_trian=2.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1128_1346_DAGGNN_node10_seed3"""

# SEED = 4

## IAF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=4, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [1:25:38<00:00,  2.14it/s, nll_train=0.5217, kl_train=0.0447, ELBO_loss=0.5664, shd_trian=21.1800, h_A=0.0006]Optimization Finished!
results/independence/normal/1128_1501_IAF_node10_seed4"""
## HF
"nan"
## DAGGNN
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=4, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [1:49:42<00:00,  1.91it/s, nll_train=24.2277, kl_train=0.0381, ELBO_loss=24.2658, shd_trian=38.0000, h_A=2.5957]Optimization Finished!
results/independence/normal/1128_2339_DAGGNN_node10_seed4"""

# SEED = 5

## IAF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=5, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [1:58:01<00:00,  1.72it/s, nll_train=4.8226, kl_train=0.2341, ELBO_loss=5.0567, shd_trian=21.3400, h_A=0.0053]Optimization Finished!
results/independence/normal/1129_0129_IAF_node10_seed5"""
## HF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=5, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [2:45:58<00:00,  1.72it/s, nll_train=0.0756, kl_train=0.0213, ELBO_loss=0.0968, shd_trian=24.7200, h_A=0.0005]Optimization Finished!
results/independence/normal/1129_0637_HF_node10_seed5"""
## DAGGNN
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=5, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299:  54% 8100/15000 [59:17<49:02,  2.34it/s, nll_train=0.0000, kl_train=0.0004, ELBO_loss=0.0004, shd_trian=20.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1129_0923_DAGGNN_node10_seed5"""

# SEED = 6

## IAF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=6, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299:  52% 7800/15000 [1:02:18<48:03,  2.50it/s, nll_train=0.0000, kl_train=0.0001, ELBO_loss=0.0002, shd_trian=8.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1129_1023_IAF_node10_seed6"""
## HF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=6, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [2:07:05<00:00,  2.20it/s, nll_train=0.0972, kl_train=0.0346, ELBO_loss=0.1318, shd_trian=22.0000, h_A=0.0005]Optimization Finished!
results/independence/normal/1129_1155_HF_node10_seed6"""
# SEED = 7

## HF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=7, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299:  36% 5400/15000 [45:10<1:10:36,  2.27it/s, nll_train=12.3709, kl_train=0.0000, ELBO_loss=12.3709, shd_trian=9.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1129_1403_HF_node10_seed7"""

## HF
"""Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=93, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299: : 17700it [2:18:26,  2.49it/s, nll_train=0.0029, kl_train=0.0013, ELBO_loss=0.0041, shd_trian=11.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1129_2354_HF_node10_seed93"""


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=1 --flow_type='IAF' --dependence_prop=0.1

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=100, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299: 100% 30000/30000 [3:09:34<00:00,  3.13it/s, nll_train=0.9308, kl_train=0.0504, ELBO_loss=0.9812, shd_trian=23.0000, h_A=0.0006]Optimization Finished!
results/independence/normal/1127_1223_IAF_node10_seed1


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=1 --flow_type='HF' --dependence_prop=0.1 # nan

In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=1 --flow_type='DAGGNN' --dependence_prop=0.1

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=100, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299:  27% 8100/30000 [42:01<2:17:19,  2.66it/s, nll_train=0.0001, kl_train=0.0003, ELBO_loss=0.0003, shd_trian=4.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1127_1629_DAGGNN_node10_seed1


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=2 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=2, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299:  38% 5700/15000 [44:17<1:03:12,  2.45it/s, nll_train=0.0000, kl_train=0.0001, ELBO_loss=0.0001, shd_trian=9.1200, h_A=0.0000]Optimization Finished!
results/independence/normal/1128_0736_IAF_node10_seed2


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=2 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=2, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0294:   8% 1195/15000 [13:15<2:13:30,  1.72it/s, nll_train=12.2751, kl_train=0.0021, ELBO_loss=12.2772, shd_trian=28.0000, h_A=0.0288]

In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=2 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=2, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299:  50% 7500/15000 [36:38<31:58,  3.91it/s, nll_train=0.0000, kl_train=0.0001, ELBO_loss=0.0001, shd_trian=1.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1128_1106_DAGGNN_node10_seed2


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=3 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50 # nan

In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=3 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=3, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [1:53:36<00:00,  2.55it/s, nll_train=0.6193, kl_train=0.0547, ELBO_loss=0.6740, shd_trian=27.0000, h_A=0.0017]Optimization Finished!
results/independence/normal/1128_1152_HF_node10_seed3


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=3 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=3, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299:  56% 8400/15000 [43:02<33:17,  3.30it/s, nll_train=0.0001, kl_train=0.0002, ELBO_loss=0.0003, shd_trian=2.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1128_1346_DAGGNN_node10_seed3


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=4 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=4, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [1:25:38<00:00,  2.14it/s, nll_train=0.5217, kl_train=0.0447, ELBO_loss=0.5664, shd_trian=21.1800, h_A=0.0006]Optimization Finished!
results/independence/normal/1128_1501_IAF_node10_seed4


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=4 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50 # nan

In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=4 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=4, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [1:49:42<00:00,  1.91it/s, nll_train=24.2277, kl_train=0.0381, ELBO_loss=24.2658, shd_trian=38.0000, h_A=2.5957]Optimization Finished!
results/independence/normal/1128_2339_DAGGNN_node10_seed4


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=5 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=5, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [1:58:01<00:00,  1.72it/s, nll_train=4.8226, kl_train=0.2341, ELBO_loss=5.0567, shd_trian=21.3400, h_A=0.0053]Optimization Finished!
results/independence/normal/1129_0129_IAF_node10_seed5


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=5 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=5, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [2:45:58<00:00,  1.72it/s, nll_train=0.0756, kl_train=0.0213, ELBO_loss=0.0968, shd_trian=24.7200, h_A=0.0005]Optimization Finished!
results/independence/normal/1129_0637_HF_node10_seed5


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=5 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=5, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=10)
Epoch: 0299:  54% 8100/15000 [59:17<49:02,  2.34it/s, nll_train=0.0000, kl_train=0.0004, ELBO_loss=0.0004, shd_trian=20.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1129_0923_DAGGNN_node10_seed5


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=6 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=6, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=10)
Epoch: 0299:  52% 7800/15000 [1:02:18<48:03,  2.50it/s, nll_train=0.0000, kl_train=0.0001, ELBO_loss=0.0002, shd_trian=8.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1129_1023_IAF_node10_seed6


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=6 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=6, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299: 100% 15000/15000 [2:07:05<00:00,  2.20it/s, nll_train=0.0972, kl_train=0.0346, ELBO_loss=0.1318, shd_trian=22.0000, h_A=0.0005]Optimization Finished!
results/independence/normal/1129_1155_HF_node10_seed6


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=7 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=7, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299:  36% 5400/15000 [45:10<1:10:36,  2.27it/s, nll_train=12.3709, kl_train=0.0000, ELBO_loss=12.3709, shd_trian=9.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1129_1403_HF_node10_seed7


In [ ]:
!python train.py --dependence_type=0 --node_size=10 --seed=93 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=10, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=93, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=10)
Epoch: 0299: : 17700it [2:18:26,  2.49it/s, nll_train=0.0029, kl_train=0.0013, ELBO_loss=0.0041, shd_trian=11.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1129_2354_HF_node10_seed93


| IAF | HF | DAGGNN |
|--|--|--|
|23.00|27.00|4.00|
|9.12|24.72|1.00|
|21.18|22.00|2.00|
|21.34|9.00|38.00|
|8.00|11.00|20.00|

평균

| IAF | HF | DAGGNN |
|--|--|--|
|16.528|18.744|13.000|

# Node 개수 : 20

In [ ]:
# SEED = 7

## IAF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=7, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299:  74% 11100/15000 [1:09:12<21:30,  3.02it/s, nll_train=0.0000, kl_train=0.0006, ELBO_loss=0.0006, shd_trian=8.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_0221_IAF_node20_seed7"""
## HF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=7, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299:  72% 10800/15000 [2:19:38<51:12,  1.37it/s, nll_train=0.0000, kl_train=0.0006, ELBO_loss=0.0006, shd_trian=28.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_0358_HF_node20_seed7"""
## DAGGNN
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=7, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299:  46% 6900/15000 [59:34<1:00:51,  2.22it/s, nll_train=0.0001, kl_train=0.0005, ELBO_loss=0.0006, shd_trian=11.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_0618_DAGGNN_node20_seed7"""

# SEED = 9

## IAF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=9, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299:  78% 11700/15000 [1:24:40<25:42,  2.14it/s, nll_train=0.0000, kl_train=0.0002, ELBO_loss=0.0002, shd_trian=5.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_1240_IAF_node20_seed9"""
## HF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=9, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299:  50% 7500/15000 [1:09:27<1:03:22,  1.97it/s, nll_train=14.8554, kl_train=0.0001, ELBO_loss=14.8555, shd_trian=22.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_2138_HF_node20_seed9"""
## DAGGNN
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=9, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299:  60% 9000/15000 [1:12:50<41:29,  2.41it/s, nll_train=0.0000, kl_train=0.0003, ELBO_loss=0.0003, shd_trian=7.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1201_0058_DAGGNN_node20_seed9"""

# SEED = 10

# IAF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=10, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [2:03:53<00:00,  2.26it/s, nll_train=25.2024, kl_train=0.0000, ELBO_loss=25.2024, shd_trian=21.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1201_0211_IAF_node20_seed10"""
# HF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=10, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [3:01:44<00:00,  1.51it/s, nll_train=25.2024, kl_train=0.0000, ELBO_loss=25.2024, shd_trian=21.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected"""
# DAGGNN
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=10, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [2:07:28<00:00,  1.94it/s, nll_train=23.1125, kl_train=1.0214, ELBO_loss=24.1340, shd_trian=9.8600, h_A=0.0007]Optimization Finished!
results/independence/normal/1201_0845_DAGGNN_node20_seed10"""

# SEED = 11

# IAF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=11, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299:  38% 5700/15000 [51:59<1:27:50,  1.76it/s, nll_train=0.0477, kl_train=0.0019, ELBO_loss=0.0496, shd_trian=36.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1201_1052_IAF_node20_seed11"""
# HF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=11, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [3:16:37<00:00,  1.43it/s, nll_train=0.1103, kl_train=0.0573, ELBO_loss=0.1676, shd_trian=66.2400, h_A=0.0002]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected"""

# SEED = 12

# HF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=12, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299: : 18600it [3:13:33,  1.71it/s, nll_train=0.0587, kl_train=0.0464, ELBO_loss=0.1051, shd_trian=51.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected"""

# DAGGNN
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=12, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299:  50% 7500/15000 [43:08<38:11,  3.27it/s, nll_train=0.0000, kl_train=0.0003, ELBO_loss=0.0003, shd_trian=15.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1202_0046_DAGGNN_node20_seed12"""

# SEED = 13

# IAF
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=13, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [1:37:38<00:00,  2.98it/s, nll_train=0.0942, kl_train=0.0217, ELBO_loss=0.1159, shd_trian=53.5000, h_A=0.0033]Optimization Finished!
results/independence/normal/1202_0202_IAF_node20_seed13"""

# DAGGNN
"""Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=23, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [1:58:00<00:00,  1.88it/s, nll_train=0.1638, kl_train=0.0395, ELBO_loss=0.2033, shd_trian=68.7000, h_A=0.0001]Optimization Finished!
results/independence/normal/1202_0435_DAGGNN_node20_seed23"""



In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=7 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=7, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299:  74% 11100/15000 [1:09:12<21:30,  3.02it/s, nll_train=0.0000, kl_train=0.0006, ELBO_loss=0.0006, shd_trian=8.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_0221_IAF_node20_seed7


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=7 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=7, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299:  72% 10800/15000 [2:19:38<51:12,  1.37it/s, nll_train=0.0000, kl_train=0.0006, ELBO_loss=0.0006, shd_trian=28.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_0358_HF_node20_seed7


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=7 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=7, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299:  46% 6900/15000 [59:34<1:00:51,  2.22it/s, nll_train=0.0001, kl_train=0.0005, ELBO_loss=0.0006, shd_trian=11.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_0618_DAGGNN_node20_seed7


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=8 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=8 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=8 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=9 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=9, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299:  78% 11700/15000 [1:24:40<25:42,  2.14it/s, nll_train=0.0000, kl_train=0.0002, ELBO_loss=0.0002, shd_trian=5.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_1240_IAF_node20_seed9


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=9 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=9, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299:  50% 7500/15000 [1:09:27<1:03:22,  1.97it/s, nll_train=14.8554, kl_train=0.0001, ELBO_loss=14.8555, shd_trian=22.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1130_2138_HF_node20_seed9


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=9 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=9, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299:  60% 9000/15000 [1:12:50<41:29,  2.41it/s, nll_train=0.0000, kl_train=0.0003, ELBO_loss=0.0003, shd_trian=7.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1201_0058_DAGGNN_node20_seed9


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=10 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=10, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [2:03:53<00:00,  2.26it/s, nll_train=25.2024, kl_train=0.0000, ELBO_loss=25.2024, shd_trian=21.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1201_0211_IAF_node20_seed10


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=10 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=10, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [3:01:44<00:00,  1.51it/s, nll_train=25.2024, kl_train=0.0000, ELBO_loss=25.2024, shd_trian=21.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=10 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=10, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [2:07:28<00:00,  1.94it/s, nll_train=23.1125, kl_train=1.0214, ELBO_loss=24.1340, shd_trian=9.8600, h_A=0.0007]Optimization Finished!
results/independence/normal/1201_0845_DAGGNN_node20_seed10


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=11 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=11, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299:  38% 5700/15000 [51:59<1:27:50,  1.76it/s, nll_train=0.0477, kl_train=0.0019, ELBO_loss=0.0496, shd_trian=36.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1201_1052_IAF_node20_seed11


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=11 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=11, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [3:16:37<00:00,  1.43it/s, nll_train=0.1103, kl_train=0.0573, ELBO_loss=0.1676, shd_trian=66.2400, h_A=0.0002]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=11 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=12 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=12 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=12, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=20)
Epoch: 0299: : 18600it [3:13:33,  1.71it/s, nll_train=0.0587, kl_train=0.0464, ELBO_loss=0.1051, shd_trian=51.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=12 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=12, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299:  50% 7500/15000 [43:08<38:11,  3.27it/s, nll_train=0.0000, kl_train=0.0003, ELBO_loss=0.0003, shd_trian=15.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1202_0046_DAGGNN_node20_seed12


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=13 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=13, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [1:37:38<00:00,  2.98it/s, nll_train=0.0942, kl_train=0.0217, ELBO_loss=0.1159, shd_trian=53.5000, h_A=0.0033]Optimization Finished!
results/independence/normal/1202_0202_IAF_node20_seed13


In [ ]:
!python train.py --dependence_type=0 --node_size=20 --seed=23 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=20, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=23, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=20)
Epoch: 0299: 100% 15000/15000 [1:58:00<00:00,  1.88it/s, nll_train=0.1638, kl_train=0.0395, ELBO_loss=0.2033, shd_trian=68.7000, h_A=0.0001]Optimization Finished!
results/independence/normal/1202_0435_DAGGNN_node20_seed23


|IAF|HF|DAGGNN|
|--|--|--|
|8.00|28.00|11.00|
|5.00|22.00|7.00|
|21.00|21.00|9.00|
|36.00|51.00|15.00|
|53.50|66.24|68.70|

평균

|IAF|HF|DAGGNN|
|--|--|--|
|24.7|37.648|22.14|

# Node 개수 : 30

In [ ]:
# a

# IAF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=11, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0299:  44% 6600/15000 [1:01:55<1:31:46,  1.53it/s, nll_train=0.0003, kl_train=0.0003, ELBO_loss=0.0006, shd_trian=7.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1202_0949_IAF_node30_seed11"""

# HF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299: 100% 15000/15000 [3:33:37<00:00,  1.33it/s, nll_train=73.9657, kl_train=0.0041, ELBO_loss=73.9698, shd_trian=168.6400, h_A=0.0002]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected"""
# DAGGNN
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1687, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  64% 9600/15000 [1:29:57<43:53,  2.05it/s, nll_train=0.0003, kl_train=0.0029, ELBO_loss=0.0031, shd_trian=12.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1203_0005_DAGGNN_node30_seed1687"""

# b

# IAF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=111, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0299:  86% 12900/15000 [2:11:32<19:30,  1.79it/s, nll_train=0.0000, kl_train=0.0005, ELBO_loss=0.0005, shd_trian=28.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1203_0135_IAF_node30_seed111"""

# HF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=111, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299: 100% 15000/15000 [3:45:30<00:00,  1.19it/s, nll_train=2.3357, kl_train=0.1123, ELBO_loss=2.4479, shd_trian=106.1400, h_A=0.0261]Optimization Finished!
results/independence/normal/1203_0639_HF_node30_seed111"""
# DAGGNN
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=111, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  42% 6300/15000 [57:06<1:07:04,  2.16it/s, nll_train=0.0000, kl_train=0.0004, ELBO_loss=0.0004, shd_trian=35.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected"""

# c

# IAF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1211, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0299:  52% 7800/15000 [1:19:34<1:23:44,  1.43it/s, nll_train=0.0000, kl_train=0.0004, ELBO_loss=0.0004, shd_trian=8.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1203_1357_IAF_node30_seed1211"""

# HF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299:  84% 12600/15000 [3:05:43<36:26,  1.10it/s, nll_train=0.0001, kl_train=0.0008, ELBO_loss=0.0009, shd_trian=39.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected"""

# DAGGNN
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  42% 6300/15000 [58:07<1:10:06,  2.07it/s, nll_train=0.0001, kl_train=0.0005, ELBO_loss=0.0006, shd_trian=12.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1204_0036_DAGGNN_node30_seed121"""

# d

# IAF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121241, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0299:  86% 12900/15000 [2:15:28<20:47,  1.68it/s, nll_train=0.2307, kl_train=0.0019, ELBO_loss=0.2326, shd_trian=52.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1204_0449_IAF_node30_seed121241"""

# HF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121241, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299: 100% 15000/15000 [2:48:25<00:00,  1.68it/s, nll_train=1.0934, kl_train=0.0660, ELBO_loss=1.1594, shd_trian=121.7000, h_A=0.0169]Optimization Finished!
results/independence/normal/1204_0729_HF_node30_seed121241"""

# DAGGNN
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121241, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  60% 9000/15000 [1:02:40<36:32,  2.74it/s, nll_train=0.0000, kl_train=0.0005, ELBO_loss=0.0005, shd_trian=22.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected"""

# e

# IAF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=21, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0169:  97% 14570/15000 [1:54:41<03:08,  2.29it/s, nll_train=0.0000, kl_train=0.0009, ELBO_loss=0.0009, shd_trian=14.0000, h_A=0.0000]"""

# HF
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=21, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299: 100% 15000/15000 [3:29:29<00:00,  1.25it/s, nll_train=0.5820, kl_train=0.0593, ELBO_loss=0.6413, shd_trian=106.0600, h_A=0.0006]Optimization Finished!
results/independence/normal/1204_1430_HF_node30_seed21"""

# DAGGNN
"""Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=21, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  56% 8400/15000 [1:11:20<49:21,  2.23it/s, nll_train=0.0000, kl_train=0.0006, ELBO_loss=0.0006, shd_trian=11.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected"""


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=11 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=11, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0299:  44% 6600/15000 [1:01:55<1:31:46,  1.53it/s, nll_train=0.0003, kl_train=0.0003, ELBO_loss=0.0006, shd_trian=7.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1202_0949_IAF_node30_seed11


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=1 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299: 100% 15000/15000 [3:33:37<00:00,  1.33it/s, nll_train=73.9657, kl_train=0.0041, ELBO_loss=73.9698, shd_trian=168.6400, h_A=0.0002]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=1687 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1687, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  64% 9600/15000 [1:29:57<43:53,  2.05it/s, nll_train=0.0003, kl_train=0.0029, ELBO_loss=0.0031, shd_trian=12.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1203_0005_DAGGNN_node30_seed1687


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=111 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=111, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0299:  86% 12900/15000 [2:11:32<19:30,  1.79it/s, nll_train=0.0000, kl_train=0.0005, ELBO_loss=0.0005, shd_trian=28.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1203_0135_IAF_node30_seed111


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=111 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=111, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299: 100% 15000/15000 [3:45:30<00:00,  1.19it/s, nll_train=2.3357, kl_train=0.1123, ELBO_loss=2.4479, shd_trian=106.1400, h_A=0.0261]Optimization Finished!
results/independence/normal/1203_0639_HF_node30_seed111


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=111 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=111, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  42% 6300/15000 [57:06<1:07:04,  2.16it/s, nll_train=0.0000, kl_train=0.0004, ELBO_loss=0.0004, shd_trian=35.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=1211 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1211, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0299:  52% 7800/15000 [1:19:34<1:23:44,  1.43it/s, nll_train=0.0000, kl_train=0.0004, ELBO_loss=0.0004, shd_trian=8.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1203_1357_IAF_node30_seed1211


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=121 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299:  84% 12600/15000 [3:05:43<36:26,  1.10it/s, nll_train=0.0001, kl_train=0.0008, ELBO_loss=0.0009, shd_trian=39.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=121 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  42% 6300/15000 [58:07<1:10:06,  2.07it/s, nll_train=0.0001, kl_train=0.0005, ELBO_loss=0.0006, shd_trian=12.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1204_0036_DAGGNN_node30_seed121


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=121241 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121241, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0299:  86% 12900/15000 [2:15:28<20:47,  1.68it/s, nll_train=0.2307, kl_train=0.0019, ELBO_loss=0.2326, shd_trian=52.0000, h_A=0.0000]Optimization Finished!
results/independence/normal/1204_0449_IAF_node30_seed121241


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=121241 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121241, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299: 100% 15000/15000 [2:48:25<00:00,  1.68it/s, nll_train=1.0934, kl_train=0.0660, ELBO_loss=1.1594, shd_trian=121.7000, h_A=0.0169]Optimization Finished!
results/independence/normal/1204_0729_HF_node30_seed121241


In [10]:
!python train.py --dependence_type=0 --node_size=30 --seed=121241 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=121241, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  60% 9000/15000 [1:02:40<36:32,  2.74it/s, nll_train=0.0000, kl_train=0.0005, ELBO_loss=0.0005, shd_trian=22.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected


In [ ]:
!python train.py --dependence_type=0 --node_size=30 --seed=21 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=21, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=30)
Epoch: 0169:  97% 14570/15000 [1:54:41<03:08,  2.29it/s, nll_train=0.0000, kl_train=0.0009, ELBO_loss=0.0009, shd_trian=14.0000, h_A=0.0000]

In [5]:
!python train.py --dependence_type=0 --node_size=30 --seed=21 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=21, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=30)
Epoch: 0299: 100% 15000/15000 [3:29:29<00:00,  1.25it/s, nll_train=0.5820, kl_train=0.0593, ELBO_loss=0.6413, shd_trian=106.0600, h_A=0.0006]Optimization Finished!
results/independence/normal/1204_1430_HF_node30_seed21


In [6]:
!python train.py --dependence_type=0 --node_size=30 --seed=21 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=30, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=21, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='DAGGNN', lagrange=1, z_size=30)
Epoch: 0299:  56% 8400/15000 [1:11:20<49:21,  2.23it/s, nll_train=0.0000, kl_train=0.0006, ELBO_loss=0.0006, shd_trian=11.0000, h_A=0.0000]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 459, in <module>
    log.flush()
OSError: [Errno 107] Transport endpoint is not connected


|IAF|HF|DAGGNN|
|--|--|--|
|7.00|168.64|12.00|
|28.00|106.14|35.00|
|8.00|39.00|12.00|
|52.00|121.70|22.00|
|14.00|106.06|11.00|

평균

|IAF|HF|DAGGNN|
|--|--|--|
|21.80|108.308|18.4|

# Node 개수 : 50

In [ ]:
# a

# IAF
"""Namespace(data_sample_size=5000, node_size=50, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1235, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=50)
Epoch: 0299: 100% 15000/15000 [3:31:23<00:00,  1.25it/s, nll_train=6.5484, kl_train=0.4926, ELBO_loss=7.0410, shd_trian=649.8600, h_A=0.3202]Optimization Finished!
results/independence/normal/1205_0012_IAF_node50_seed1235"""
# HF

# DAGGNN

In [5]:
!python train.py --dependence_type=0 --node_size=50 --seed=1235 --flow_type='IAF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=50, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=1235, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='IAF', lagrange=1, z_size=50)
Epoch: 0299: 100% 15000/15000 [3:31:23<00:00,  1.25it/s, nll_train=6.5484, kl_train=0.4926, ELBO_loss=7.0410, shd_trian=649.8600, h_A=0.3202]Optimization Finished!
results/independence/normal/1205_0012_IAF_node50_seed1235


In [6]:
!python train.py --dependence_type=0 --node_size=50 --seed=21 --flow_type='HF' --dependence_prop=0.1 --k_max_iter=50

Namespace(data_sample_size=5000, node_size=50, graph_degree=2, graph_dist='normal', graph_scale=1.0, graph_mean=0.0, graph_linear_type='linear', dependence_type=0, dependence_prop=0.1, optimizer='Adam', graph_threshold=0.3, tau_A=0.0, lambda_A=0.0, c_A=1, use_A_connect_loss=0, use_A_positiver_loss=0, cuda=0, seed=21, epochs=300, batch_size=100, lr=0.003, encoder_hidden=64, decoder_hidden=64, temp=0.5, k_max_iter=50, encoder_dropout=0.0, decoder_dropout=0.0, lr_decay=200, gamma=1.0, h_tol=1e-08, x_dims=1, z_dims=1, number_of_flows=5, flow_type='HF', lagrange=1, z_size=50)
Epoch: 0253:  62% 9254/15000 [3:20:25<2:24:30,  1.51s/it, nll_train=3.2447, kl_train=1.4170, ELBO_loss=4.6617, shd_trian=788.2800, h_A=0.1146]Traceback (most recent call last):
  File "/content/drive/MyDrive/DL_Stat/NFG-VAE/train.py", line 421, in <module>
    ELBO_loss, NLL_loss, MSE_loss, graph, origin_A, LT = train(epoch=epoch, model=vae, best_val_loss=best_ELBO_loss, G=G, lambda_A=lambda_A, c_A=c_A, optimizer=optim

In [7]:
!python train.py --dependence_type=0 --node_size=50 --seed=21 --flow_type='DAGGNN' --dependence_prop=0.1 --k_max_iter=50

shell-init: error retrieving current directory: getcwd: cannot access parent directories: Transport endpoint is not connected
shell-init: error retrieving current directory: getcwd: cannot access parent directories: Transport endpoint is not connected
python3: can't open file 'train.py': [Errno 107] Transport endpoint is not connected
